plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryParentProjectId.java

```diff
@@ -0,0 +1,116 @@
+
+import static io.trino.testing.MaterializedResult.DEFAULT_PRECISION;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingProperties.requiredNonEmptySystemProperty;
+import static io.trino.tpch.TpchTable.NATION;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TestBigQueryParentProjectId
+        extends AbstractTestQueryFramework
+{
+    private final String testingProjectId;
+    private final String testingParentProjectId;
+    private DistributedQueryRunner queryRunner;
+
+    TestBigQueryParentProjectId()
+    {
+        testingProjectId = requiredNonEmptySystemProperty("testing.bigquery-project-id");
+        testingParentProjectId = requiredNonEmptySystemProperty("testing.bigquery-parent-project-id");
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        queryRunner = BigQueryQueryRunner.builder()
+                .setConnectorProperties(ImmutableMap.<String, String>builder()
+                        .put("bigquery.project-id", testingProjectId)
+                        .put("bigquery.parent-project-id", testingParentProjectId)
+                        .buildOrThrow())
+                .setInitialTables(ImmutableList.of(NATION))
+                .build();
+        return queryRunner;
+    }
+
+    @Test
+    void testQueriesWithParentProjectId()
+            throws Exception
+    {
+        // tpch schema is available in both projects
+        assertThat(computeScalar("SELECT name FROM bigquery.tpch.nation WHERE nationkey = 0")).isEqualTo("ALGERIA");
+        assertThat(computeScalar("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM tpch.nation WHERE nationkey = 0'))")).isEqualTo("ALGERIA");
+        assertThat(computeScalar(format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM %s.tpch.nation WHERE nationkey = 0'))", testingProjectId)))
+                .isEqualTo("ALGERIA");
+        assertThat(computeScalar(format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM %s.tpch.nation WHERE nationkey = 0'))", testingParentProjectId)))
+                .isEqualTo("ALGERIA");
+
+        String trinoSchema = "someschema_" + randomNameSuffix();
+        try (AutoCloseable ignored = withSchema(trinoSchema); TestTable table = newTrinoTable("%s.test_create_table_already_exists".formatted(trinoSchema), "(col1 INT)")) {
+            String tableName = table.getName().split("\\.")[1];
+            // schema created in testingProjectId is present in testingProjectId and NOT present in testingParentProjectId
+            assertThat(computeActual(format(
+                    "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT schema_name FROM `%s.region-us.INFORMATION_SCHEMA.SCHEMATA`'))",
+                    testingProjectId)))
+                    .contains(new MaterializedRow(DEFAULT_PRECISION, trinoSchema));
```

Comment:
You could use `private static MaterializedRow row(String value)` here

Reply:
Oh, sure thing!